MapboxNavigation/InstructionsBannerView.swift

```diff
@@ -114,17 +118,14 @@ open class BaseInstructionsBannerView: UIControl {
     public func update(for currentLegProgress: RouteLegProgress) {
         let stepProgress = currentLegProgress.currentStepProgress
         let distanceRemaining = stepProgress.distanceRemaining
+        let visualInstructionUpdated = currentStepIndex == currentLegProgress.stepIndex && currentVisualInstructionIndex == stepProgress.visualInstructionIndex
         
-        guard let visualInstructions = stepProgress.remainingVisualInstructions else { return }
+        guard let visualInstruction = currentLegProgress.currentStepProgress.currentVisualInstruction else { return }
         
-        for visualInstruction in visualInstructions {
-            if stepProgress.distanceRemaining <= visualInstruction.distanceAlongStep || stepProgress.visualInstructionIndex == 0 {
-                
-                set(visualInstruction)
-                
-                stepProgress.visualInstructionIndex += 1
-                break
-            }
+        if !visualInstructionUpdated {
```

Comment:
This should probably read, `if updated, do update`. Not `if not updated, do update.`